keps/487-kubectl-plugin-listing/README.md

```diff
@@ -0,0 +1,316 @@
+template.
+Ensure the TOC is wrapped with
+  <code>&lt;!-- toc --&rt;&lt;!-- /toc --&rt;</code>
+tags, and then generate with `hack/update-toc.sh`.
+-->
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+- [Proposal](#proposal)
+  - [User Stories (Optional)](#user-stories-optional)
+    - [Story 1](#story-1)
+    - [Story 2](#story-2)
+  - [Risks and Mitigations](#risks-and-mitigations)
+- [Design Details](#design-details)
+  - [Version](#version)
+  - [Describe](#describe)
+  - [List](#list)
+  - [Stop](#stop)
+  - [Test Plan](#test-plan)
+      - [Prerequisite testing updates](#prerequisite-testing-updates)
+    - [Unit Tests](#unit-tests)
+    - [Integration tests](#integration-tests)
+  - [Graduation Criteria](#graduation-criteria)
+- [Implementation History](#implementation-history)
+- [Drawbacks](#drawbacks)
+- [Alternatives](#alternatives)
+<!-- /toc -->
+
+## Summary
+
+Provide an easy way to view and manage kueue abstractions via command line tool.
+
+## Motivation
+
+Create tools that provide useful information in an intuitive way for cluster users and could be later extended as managing tool for cluster administrators.
+As a cluster user I don’t need to dive into the difference between job and Workload concept in kueue and should be able quickly discover the job/Workload and ClusterQueue status or see the error message what is misconfigured/should be fixed.
+
+### Goals
+
+Impement basic kueue CLI commands that was described in PRD:
+
+- Display the Workload status.
+- Describe Workload.
+- List Workloads in specific queue/namespace.
+- List admitted/pending workloads
+- List all ClusterQueues.
+- Describe ClusterQueue.
+- Stop Workload.
+- Stop ClusterQueue.
+
+### Non-Goals
+
+## Proposal
+
+We propose creating a Kubectl plugin that exposes kueue specific information. The main command-line interactions will take the following shape:
+
+```bash
+$kubectl kueue <type> <command> <name> <flags> <options>
+```
+
+Set of supported commands: `list`, `describe`, `stop`.
+Set of supported types: `workload`, `clusterqueue`, `localqueue`.
+Sets of supported flags are different for each command/type.
+
+The plugin could be downloaded directly from the repo or installed by the Krew plugging manager.
+
+### User Stories (Optional)
+
+#### Story 1
+
+As a user submitting workloads, I want to easily see the status of an entire Workload, or check on why my workload is not being admitted. I can do this with the new proposed plugin.
+
+```bash
+# List workloads in the cluster.
+$kubectl kueue workloads get
+# Describe particular workload.
+$kubeclt kueue workload describe <name>
+# Display state and conditions to see what could be wrong with workload.
+$kubeclt kueue workload describe <name> --status
+
+```
+
+#### Story 2
+
+As a cluster admin I want to have an easy way to manage cluster queues:
+
+```bash
+$kubectl kueue clusterqueue stop <queue-name>
```

Comment:
Please clarify the steps needed for the command "before" the KEP.

Reply:
What do you mean? 
We need an API first, this currently in progress https://github.com/kubernetes-sigs/kueue/pull/1288

Reply:
Added links to the issues